uamqp/client.py

```diff
@@ -640,13 +640,24 @@ def receive_message_batch(self, max_batch_size=None, on_message_received=None, t
             return batch
 
         while receiving and not expired and len(batch) < max_batch_size:
-            receiving = self.do_work(batch=max_batch_size)
+
```

Comment:
Is this new section of code migrated over from v1?
Seems we would also need it in the async version?
We are checking `qsize` a lot - I wonder how expensive this is....

Reply:
I have improved the algorithm to call qsize exactly once in each iteration.
let me know how the new algorithm looks to you.

Reply:
Yeah I think this looks better. Would be interesting to do a `timeit` comparison.

Reply:
I have written a small "timeit" script result to test it.

based on my scripts, the original one is slight faster resulting in ~1.5 seconds, with the change it's ~1.6 seconds. So it's like a roughly ~6% performance loss after adding the extra logic to enable immediate return in receive message batch method.

I've uploaded the timeit script, let me know if you want me to tweak the script a bit further.